server/resp/src/test/java/org/infinispan/server/resp/JsonCommandsTest.java

```diff
@@ -0,0 +1,325 @@
+      result = redis.jsonGet(key, jp);
+      assertThat(result).hasSize(1);
+      assertThat(compareJSON(result.get(0), jv)).isEqualTo(true);
+
+      // Test adding a field in a leaf
+      jp = new JsonPath("$.key.key2");
+      jv = result.get(0);
+      JsonValue jv1 = new DefaultJsonParser().createJsonValue("{\"key2\":\"value2\"}");
+      assertThat(redis.jsonSet(key, jp, jv1)).isEqualTo("OK");
+      result = redis.jsonGet(key);
+      assertThat(result).hasSize(1);
+      assertThat(compareJSONSet(result.get(0), jv, "$.key.key2", jv1)).isEqualTo(true);
+
+   }
+
+   @Test
+   public void testJSONSETWrongType() {
+      String key = k();
+      JsonValue jv = new DefaultJsonParser().createJsonValue("{\"key\":\"value\"}");
+      JsonPath jp = new JsonPath("$");
+
+      // Check get a string as json
+      assertWrongType(() -> redis.set(key, v()), () -> redis.jsonGet(key, new JsonPath("$")));
+      // Check set an existing string
+      assertWrongType(() -> {} , () -> redis.jsonSet(key, jp, jv));
+      // Check with non root
+      String k1 = k(1);
+      assertWrongType(() -> redis.set(k1, v()), () -> redis.jsonGet(k1, new JsonPath("$.k1")));
+      // Check json is not a string
+      String k2 = k(2);
+      assertWrongType(() -> redis.jsonSet(k2, jp, jv), () -> redis.get(k2));
+   }
+
+   @Test
+   public void testJSONSETWithPath() {
+      String key = k();
+      // Create json
+      JsonPath jpRoot = new JsonPath("$");
+      JsonValue jvDoc = new DefaultJsonParser().createJsonValue("{\"root\": { \"k1\" : \"v1\", \"k2\":\"v2\"}}");
+      assertThat(redis.jsonSet(key, jpRoot, jvDoc)).isEqualTo("OK");
+
+      // Modify json
+      JsonPath jpLeaf = new JsonPath("$.root.k1");
+      JsonValue jvNew = new DefaultJsonParser().createJsonValue("\"newv1\"");
+      assertThat(redis.jsonSet(key, jpLeaf, jvNew)).isEqualTo("OK");
+
+      // Verify
+      var result = redis.jsonGet(key, jpRoot);
+      assertThat(result).hasSize(1);
+      assertThat(compareJSONSet(result.get(0), jvDoc, "$.root.k1", jvNew));
+   }
+
+   @Test
+   public void testJSONSETWithPathMulti() {
+      String key = k();
+      // Create json
+      JsonPath jpRoot = new JsonPath("$");
+      JsonValue jvDoc = new DefaultJsonParser().createJsonValue(
+            """
+                  {"r1": { "k1" : "v1", "k2":"v2"}, "r2": { "k1" : "v1", "k2": "v2"}}"
+                  """);
+      assertThat(redis.jsonSet(key, jpRoot, jvDoc)).isEqualTo("OK");
+
+      // Modify json
+      JsonPath jpLeaf = new JsonPath("$..k1");
+      JsonValue jvNew = new DefaultJsonParser().createJsonValue("\"newv1\"");
+      assertThat(redis.jsonSet(key, jpLeaf, jvNew)).isEqualTo("OK");
+
+      // Verify
+      var result = redis.jsonGet(key);
+      assertThat(result).hasSize(1);
+      assertThat(compareJSONSet(result.get(0), jvDoc, "$..k1", jvNew)).isEqualTo(true);
+   }
+
+   @Test
+   public void testJSONSETWithXX() {
```

Comment:
This seems to be missing a few other combinations. XX when the document exists. And also NX.